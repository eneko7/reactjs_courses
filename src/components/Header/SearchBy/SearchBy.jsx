import React from 'react';
import SortInput from '../../SortInput';
import styles from './SearchBy.scss';

const SearchBy = () => (
  <div className={styles.searchBy_wrapper}>
    <span className={styles.searchBy_title}>SEARCH BY</span>
    <SortInput
      name="searchBy"
      id="TITLE"
      text="TITLE"
      inputClass="searchType_input"
      labelClass="searchType_button"
      defaultChecked="defaultChecked"
    />
    <SortInput
      name="searchBy"
      id="GENRE"
      text="GENRE"
      inputClass="searchType_input"
      labelClass="searchType_button"
    />
  </div>
);

export default SearchBy;
