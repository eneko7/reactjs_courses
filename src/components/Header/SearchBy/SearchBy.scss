@import "../../../assets/constants/constants.scss";

.search_by_wrapper {
  position: relative;
  z-index: inherit;
  margin-top: 30px;
  width: 30%;
  display: flex;
  justify-content: space-between;

  .search_by_title {
    color: #fff;
    font-family: $fontFamilyBold;
    font-size: 15px;
    line-height: 30px;
  }
}

@media (max-width: $desktop) {
  .search_by_wrapper {
    width: 40%;
  }
}

@media (max-width: $laptop) {
  .search_by_wrapper {
    width: 37%;
  }
}

@media (max-width: $largeTablet) {
  .search_by_wrapper {
    width: 45%;
  }
}

@media (max-width: $tablet) {
  .search_by_wrapper {
    width: 65%;
  }
}

@media (max-width: $phone) {
  .search_by_wrapper {
    width: 100%;
  }
}
