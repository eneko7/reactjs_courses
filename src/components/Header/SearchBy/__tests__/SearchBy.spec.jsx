import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import SearchBy from '../SearchBy';

const renderer = new ShallowRenderer();
renderer.render(<SearchBy />);
describe('SearchBy', () => {
  it('renders correctly', () => {
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
