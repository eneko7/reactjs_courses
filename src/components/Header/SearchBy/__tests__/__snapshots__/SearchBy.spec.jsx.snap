// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBy renders correctly 1`] = `
<div
  className="searchBy_wrapper"
>
  <span
    className="searchBy_title"
  >
    SEARCH BY
  </span>
  <SortInput
    defaultChecked="defaultChecked"
    id="TITLE"
    inputClass="searchType_input"
    labelClass="searchType_button"
    name="searchBy"
    text="TITLE"
  />
  <SortInput
    defaultChecked=""
    id="GENRE"
    inputClass="searchType_input"
    labelClass="searchType_button"
    name="searchBy"
    text="GENRE"
  />
</div>
`;
